server_src/constants/state.constants.js

```diff
@@ -130,38 +95,155 @@ const OP_STATE = {
+  },
+  [OP_STATE.OpeningSerial]: {
+    state: PSTATE.Searching, // Lack of better
+    desc: "Your printer is connecting to serial"
+  },
+  [OP_STATE.DetectingSerial]: {
+    state: PSTATE.Searching, // Lack of better
+    desc: "Your printer is detecting serial connections"
+  },
+  [OP_STATE.Connecting]: {
+    state: PSTATE.Connecting,
+    desc: "Your printer is connecting to serial"
+  },
+  [OP_STATE.Operational]: {
+    state: PSTATE.Operational,
+    desc: "Printer is ready to print"
+  },
+  [OP_STATE.StartingPrintFromSD]: {
+    state: PSTATE.Searching,
+    desc: "STARTING PRINT FROM SD!"
+  },
+  [OP_STATE.StartSendingPrintToSD]: {
+    state: PSTATE.Searching,
+    desc: "Starting to send file to SD"
+  },
+  [OP_STATE.Starting]: {
+    state: PSTATE.Starting,
+    desc: "Printing right now"
+  },
+  [OP_STATE.TransferringFileToSD]: {
+    state: PSTATE.Searching,
+    desc: "Transferring to SD"
+  },
+  [OP_STATE.SendingFileToSD]: {
+    state: PSTATE.Searching,
+    desc: "Busy sending file to SD"
+  },
+  [OP_STATE.PrintingFromSD]: {
+    state: PSTATE.Printing,
+    desc: "PRINTING FROM SD!"
+  },
+  [OP_STATE.Printing]: {
+    state: PSTATE.Printing,
+    desc: "Printing right now"
+  },
+  [OP_STATE.Cancelling]: {
+    state: PSTATE.Cancelling,
+    desc: "Print is cancelling"
+  },
+  [OP_STATE.Pausing]: {
+    state: PSTATE.Pausing,
+    desc: "Printing paused"
+  },
+  [OP_STATE.Paused]: {
+    state: PSTATE.Paused,
+    desc: "Printing paused"
+  },
+  [OP_STATE.Resuming]: {
+    state: PSTATE.Starting,
+    desc: "Print resuming"
+  },
+  [OP_STATE.Finishing]: {
+    state: PSTATE.Complete,
+    desc: "Print finishing"
+  },
+  [OP_STATE.UnknownState]: {
+    state: "Unknown state",
+    desc: "Unknown state"
+  }
+};
+
+function remapOctoPrintState(octoPrintState) {
+  // Handy stuff!
+  const flags = octoPrintState.flags;
+  const stateLabel = octoPrintState.text;
+
+  if (stateLabel.includes("Error:") || stateLabel.includes("error")) {
+    return {
+      state: PSTATE.Error,
+      flags,
+      desc: stateLabel
+    };
+  }
+
+  const mapping = OF_STATE_REMAP[stateLabel];
+  mapping.flags = flags;
+  if (!!mapping) return mapping;
+
+  return {
+    state: stateLabel,
+    flags,
+    desc: "OctoPrint's state was not recognized"
+  };
+}
+
 const mapStateToColor = (state) => {
   if (state === PSTATE.Loading) {
     return { name: "dark", hex: "#262626", category: CATEGORY.Idle };
   }
   if (state === PSTATE.Operational) {
     return { name: "dark", hex: "#262626", category: CATEGORY.Idle };
   }
+  if (state === PSTATE.Online) {
+    return { name: "success", hex: "#00330e", category: CATEGORY.Idle };
+  }
   if (state === PSTATE.Paused) {
     return { name: "warning", hex: "#583c0e", category: CATEGORY.Idle };
   }
   if (state === PSTATE.Printing) {
-    return { name: "warning", hex: "#583c0e", category: CATEGORY.Active };
+    return { name: "success", hex: "#583c0e", category: CATEGORY.Active };
```

Comment:
This should be "warning" sucess for printing states is only used for a print completed. Think it's confusing to the user those both been the same.

Reply:
I do agree, wasnt supposed to be in this PR. Good catch. I do think we should not be limited to the limited set of colors by bootstrap. (Also given the Contrast Ratio test failing). 

Will revert

Reply:
I agree with that totally.

Reply:
Reverted